views/audit/big_query_audit_log_v2.sql

```diff
@@ -0,0 +1,759 @@
+          '$.jobChange.job.jobConfig.loadConfig.createDisposition')) 
+      AS loadcreateDisposition,
+      COALESCE(
+        JSON_EXTRACT_SCALAR(protopayload_auditlog.metadataJson,
+          '$.jobInsertion.job.jobConfig.loadConfig.writeDisposition'),
+        JSON_EXTRACT_SCALAR(protopayload_auditlog.metadataJson,
+          '$.jobChange.job.jobConfig.loadConfig.writeDisposition')) 
+      AS loadwriteDisposition,
+      COALESCE(
+        JSON_EXTRACT_SCALAR(protopayload_auditlog.metadataJson,
+          '$.jobInsertion.job.jobConfig.loadConfig.schemaJson'),
+        JSON_EXTRACT_SCALAR(protopayload_auditlog.metadataJson,
+          '$.jobChange.job.jobConfig.loadConfig.schemaJson')) 
+      AS loadschemaJson,
+      COALESCE(
+        JSON_EXTRACT_SCALAR(protopayload_auditlog.metadataJson,
+          '$.jobInsertion.job.jobConfig.loadConfig.destinationTable'),
+        JSON_EXTRACT_SCALAR(protopayload_auditlog.metadataJson,
+          '$.jobChange.job.jobConfig.loadConfig.destinationTable')) AS loadDestinationTable,
+      COALESCE(
+        JSON_EXTRACT_SCALAR(protopayload_auditlog.metadataJson,
+          '$.jobInsertion.job.jobConfig.loadConfig.destinationTableEncryption.kmsKeyName'),
+        JSON_EXTRACT_SCALAR(protopayload_auditlog.metadataJson,
+          '$.jobChange.job.jobConfig.loadConfig.destinationTableEncryption.kmsKeyName')
+      ) AS loadkmsKeyName,
+      JSON_EXTRACT_SCALAR(protopayload_auditlog.metadataJson,'$.jobInsertion.job.jobConfig.loadConfig.load'),
+      /*Queries related to queryConfig job
+      https://cloud.google.com/bigquery/docs/reference/auditlogs/rest/Shared.Types/BigQueryAuditMetadata#query */
+      COALESCE(
+        JSON_EXTRACT_SCALAR(protopayload_auditlog.metadataJson,
+          '$.jobInsertion.job.jobConfig.queryConfig.queryTruncated'),
+        JSON_EXTRACT_SCALAR(protopayload_auditlog.metadataJson,
+          '$.jobChange.job.jobConfig.queryConfig.queryTruncated')) 
+      AS queryTruncated,
+      COALESCE(
+        JSON_EXTRACT_SCALAR(protopayload_auditlog.metadataJson,
+          '$.jobInsertion.job.jobConfig.queryConfig.schemaJsonUrisTruncated'),
+        JSON_EXTRACT_SCALAR(protopayload_auditlog.metadataJson,
+          '$.jobChange.job.jobConfig.queryConfig.schemaJsonUrisTruncated')) 
+      AS queryschemaJsonTruncated,
+      COALESCE(
+        JSON_EXTRACT_SCALAR(protopayload_auditlog.metadataJson,
+          '$.jobInsertion.job.jobConfig.queryConfig.createDisposition'),
+        JSON_EXTRACT_SCALAR(protopayload_auditlog.metadataJson,
+          '$.jobChange.job.jobConfig.queryConfig.createDisposition')) 
+      AS querycreateDisposition,
+      COALESCE(
+        JSON_EXTRACT_SCALAR(protopayload_auditlog.metadataJson,
+          '$.jobInsertion.job.jobConfig.queryConfig.tableDefinitions.name'),
+        JSON_EXTRACT_SCALAR(protopayload_auditlog.metadataJson,
+          '$.jobChange.job.jobConfig.queryConfig.tableDefinitions.name')) 
+      AS queryName,
```

Comment:
Two points here:
- description from our online docs for tableDefinitions.name is "Name of the table, used in queries." so your field name should represent similar
- tableDefinitions is an array and you're not accounting for that here, properly parse the json string you extract and split by commas into an array.

Reply:
There are many other fields throughout that are arrays but you're treating them as strings. Please address them similarly